core/request.py

```diff
@@ -310,48 +315,81 @@ def processRequest(self):
 			except Exception as e:
 				logging.exception(e)
 				raise
-		except errors.HTTPException as e:
-			if conf["viur.debug.traceExceptions"]:
-				raise
-			self.response.body = b""
-			self.response.status = '%d %s' % (e.status, e.name)
-			res = None
-			if conf["viur.errorHandler"]:
-				try:
-					res = conf["viur.errorHandler"](e)
-				except Exception as newE:
-					logging.error("viur.errorHandler failed!")
-					logging.exception(newE)
-					res = None
-			if not res:
-				tpl = Template(open(conf["viur.errorTemplate"], "r").read())
-				res = tpl.safe_substitute({"error_code": e.status, "error_name": e.name, "error_descr": e.descr})
-			self.response.write(res.encode("UTF-8"))
-		except Exception as e:  # Something got really wrong
-			logging.error("Viur caught an unhandled exception!")
-			logging.exception(e)
+		except Exception as e:
+			if isinstance(e, errors.HTTPException):
+				if conf["viur.debug.traceExceptions"]:
+					raise
+				errorCode = e.status
+				errorName = e.name
+				errorDescr = e.descr
+			else:
+				# Something got really wrong
+				logging.error("Viur caught an unhandled exception!")
+				logging.exception(e)
+				errorCode = 500
+				errorName = "Internal Server Error"
+				errorDescr = "The server encountered an unexpected error and is unable to process your request."
+
 			self.response.body = b""
-			self.response.status = 500
+			self.response.status = f"{errorCode} {errorName}"
+
 			res = None
 			if conf["viur.errorHandler"]:
 				try:
 					res = conf["viur.errorHandler"](e)
 				except Exception as newE:
 					logging.error("viur.errorHandler failed!")
 					logging.exception(newE)
-					res = None
+
 			if not res:
-				tpl = Template(open(conf["viur.errorTemplate"], "r").read())
-				descr = "The server encountered an unexpected error and is unable to process your request."
-				if self.isDevServer:  # Were running on development Server
-					strIO = StringIO()
-					traceback.print_exc(file=strIO)
-					descr = strIO.getvalue()
-					descr = descr.replace("<", "&lt;").replace(">", "&gt;").replace(" ", "&nbsp;").replace("\n",
-																										   "<br />")
-				res = tpl.safe_substitute(
-					{"error_code": "500", "error_name": "Internal Server Error", "error_descr": descr})
+				with open(conf["viur.errorTemplate"], "r") as fh:
+					tpl = Template(fh.read())
+				debugInformation = ""
+				# We're running on development server or verbose mode is on
+				if self.isDevServer or conf["viur.debug.verboseErrorHandler"]:
+					if not isinstance(e, errors.HTTPException):
```

Comment:
@viur-framework/developers What do your think about this. Shall we print a traceback on `HTTPException`s or not? Might be helpful to find the right case, if there are multiple reasons.